Source/Modules/python.cxx

```diff
@@ -4345,6 +4352,146 @@ class PYTHON:public Language {
     Printf(f, "  }\n");
     Printv(f, "#endif\n", NIL);
     Printf(f, "};\n\n");
+    Printf(f, "static PyTypeObject* create_%s_type(PyTypeObject* type, PyTypeObject** bases, PyObject* dict) {\n", templ);
+    Printv(f, "  PyObject* tuple_bases;\n", NIL);
+    Printf(f, "  PyTypeObject* pytype = (PyTypeObject*)&%s_type;\n", templ);
+    Printf(f, "  pytype->tp_dict = dict;\n");
+    Printv(f, "  SwigPyBuiltin_SetMetaType(pytype, type);\n", NIL);
+    static String *tp_new = NewString("PyType_GenericNew");
+    Printf(f, "  pytype->tp_new = %s;\n", getSlot(n, "feature:python:tp_new", tp_new));
+    Printv(f, "  tuple_bases = SwigPyBuiltin_InitBases(bases);\n", NIL);
+    Printv(f, "  pytype->tp_base = bases[0];\n", NIL);
+    Printv(f, "  Py_INCREF(pytype->tp_base);\n", NIL);
+    Printv(f, "  pytype->tp_bases = tuple_bases;\n", NIL);
+    Printv(f, "  if (PyType_Ready(pytype) < 0) {\n", NIL);
+    Printf(f, "    PyErr_SetString(PyExc_TypeError, \"Could not create type '%s'.\");\n", symname);
+    Printv(f, "    return NULL;\n", NIL);
+    Printv(f, "  }\n", NIL);
+    Printf(f, "  return pytype;\n");
+    Printf(f, "}\n\n");
+
+    Printv(f, "#else\n", NIL);
+
+    Printf(f, "static PyTypeObject* create_%s_type(PyTypeObject* type, PyTypeObject** bases, PyObject* dict) {\n", templ); 
+    Printf(f, "  PyMemberDef members[] = {\n");
+    Printf(f, "  {(char*)\"__dictoffset__\", T_PYSSIZET, %s, READONLY, NULL},\n", getSlot(n, "feature:python:tp_dictoffset", tp_dictoffset_default));
+    //Printf(f, "  {(char*)\"__weaklistoffset__\", T_PYSSIZET, %s, READONLY, NULL},\n", getSlot(n, "feature:python:tp_weaklistoffset"));
+    //Printf(f, "  {(char*)\"__vectorcalloffset__\", T_PYSSIZET, %s, READONLY, NULL},\n", getSlot(n, "feature:python:tp_vectorcall_offset"));
```

Comment:
All these commented out lines of code don't seem helpful - if they need resolving, it'd be best to actually resolve them, and failing that at least document with code comments why they haven't been resolved and any implications of that.  If there's nothing to resolve it doesn't seem useful to include them.

Reply:
Right, they aren't helpful.  I've removed them because they only make sense if the C-struct has those members.  The only member in the C-struct right now is a dictionary.